src/main/java/org/opensearch/ad/model/AnomalyResult.java

```diff
@@ -89,42 +98,107 @@
-    public AnomalyResult(
-        String detectorId,
-        Double anomalyScore,
-        Double anomalyGrade,
-        Double confidence,
-        List<FeatureData> featureData,
-        Instant dataStartTime,
-        Instant dataEndTime,
-        Instant executionStartTime,
-        Instant executionEndTime,
-        String error,
-        User user,
-        Integer schemaVersion
-    ) {
-        this(
-            detectorId,
-            anomalyScore,
-            anomalyGrade,
-            confidence,
-            featureData,
-            dataStartTime,
-            dataEndTime,
-            executionStartTime,
-            executionEndTime,
-            error,
-            null,
-            user,
-            schemaVersion
-        );
-    }
+    // sequence index (the number of updates to RCF) -- it is possible in imputation
+    // that
+    // the number of updates more than the input tuples seen by the overall program
+    private final Long totalUpdates;
+
+    // flag indicating if the anomaly is the start of an anomaly or part of a run of
+    // anomalies
+    private final Boolean startOfAnomaly;
+
+    // flag indicating if the time stamp is in elevated score region to be
+    // considered as anomaly
+    private final Boolean inHighScoreRegion;
+
+    /**
+     * position of the anomaly vis a vis the current time (can be -ve) if anomaly is
+     * detected late, which can and should happen sometime; for shingle size 1; this
+     * is always 0
+     */
+    private final Integer relativeIndex;
+
+    // a flattened version denoting the basic contribution of each input variable
+    private final double[] currentTimeAttribution;
 
+    /*
+    oldValues is related to relativeIndex and startOfAnomaly.
+    So if we detect anomaly late, we get the baseDimension values from the past (current is 0).
+    That is, we look back relativeIndex * baseDimensions.
+    
+    For example, current shingle is
+    "currentValues": [
+    6819.0,
+    2375.3333333333335,
+    0.0,
+    49882.0,
+    92070.0,
+    5084.0,
+    2072.809523809524,
+    0.0,
+    43529.0,
+    91169.0,
+    8129.0,
+    2582.892857142857,
+    12.0,
+    54241.0,
+    84596.0,
+    11174.0,
+    3092.9761904761904,
+    24.0,
+    64952.0,
+    78024.0,
+    14220.0,
+    3603.059523809524,
+    37.0,
+    75664.0,
+    71451.0,
+    17265.0,
+    4113.142857142857,
+    49.0,
+    86376.0,
+    64878.0,
+    16478.0,
+    3761.4166666666665,
+    37.0,
+    78990.0,
+    70057.0,
+    15691.0,
+    3409.690476190476,
+    24.0,
+    71604.0,
+    75236.0
+    ],
+    Since relativeIndex is -2, we look back baseDimension * 2 and get the oldValues:
+    "startOfAnomaly": true,
+    "inHighScoreRegion": true,
+    "relativeIndex": -2,
+    "oldValues": [
+    17265.0,
+    4113.142857142857,
+    49.0,
+    86376.0,
+    64878.0
+    ],
+    
+    So oldValues is null when relativeIndex is 0 or startOfAnomaly is true.*/
+    private final double[] oldValues;
```

Comment:
Got it, thanks for clarifying, makes sense. So seems like `oldValues` is more of just an internal way for storing "what values caused this anomaly", but not a cx-friendly value, whereas feature attribution and expected values are cx-friendly, and may possibly use `oldValues` as part of their input when determining their own values.

Reply:
feature attribution and expected values may not be cx-friendly too as they may correspond to oldValues not  input values.

Reply:
Re "is the expected value calculated based on individual data point rather than a whole shingle? Feel confused that we tell a detector interval is anomalous or not based on shingle, but we calculate expected value/attribution based on individual data point (old value)?" ---> it's based on the shingle. In fact it is much harder to predict based on a single value :) without the context. That is why the relative position of *where in the shingle are we most likely to have a problem* is needed (relativeIndex).

Reply:
So the "relevant" values should refer to that time slice. For most simple anomalies this is often 0. But there are very good reasons why it's hard to have a generic detector with 0 time lag (and not have a lot of mistakes).